manager-bundle/src/ContaoManager/Plugin.php

```diff
@@ -342,6 +344,50 @@ private function checkMailerTransport(array $extensionConfigs, ContainerBuilder
         return $extensionConfigs;
     }
 
+    /**
+     * Adds a default orm mapping for the App namespace if none was configured.
+     *
+     * @return array<string,array<string,array<string,array<string,mixed>>>>
+     */
+    private function addDefaultDoctrineMapping(array $extensionConfigs, ContainerBuilder $container): array
+    {
+        // Do not add an 'App' mapping if it already exists or any existing
+        // mapping is targeting the '%kernel.project_dir%/src/Entity' directory.
+        foreach ($extensionConfigs as $config) {
+            if (isset($config['orm']['mappings']['App'])) {
```

Comment:
I would actually not add any default config if **any** mapping is defined in the configuration.

Reply:
I thought about this as well. But couldn't that interfere with 3rd party bundles defining mappings?

Reply:
They shouldn't be registered in the bundle configuration, I would think? Or do you know what's best practice for a bundle to register entities?

Reply:
After reading some code I realized two things:
* Mappings can also be defined for a certain entity manager.
* The `DoctrineExtension` (doctrine-bundle) configures mappings for all bundles under the default manager like so:
```yml
doctrine:
    orm:
        entity_managers:
            default:
                mappings:
                    XYZBundle:
                        mapping: true
                        is_bundle: true
                    [...]              
```

&rarr; Bundle entities are getting mapped automatically if they live in `<bundle_src>/Entity` but bundles with a special config could derive from this. In Sf land they would create a compiler pass based on the `DoctrineOrmMappingsPass` (doctrine-bundle). In Contao land they could also prepend config to do so. I had done the latter as well in some bundles quite some time ago :smile: ...  I'm pretty sure at some point earlier on the bundle entities did not get mapped automatically.

Note, that all but the prepended config (via a Contao Manager plugin) cannot be read at this point as they are processed inside a DI configuration. 

So, this implies following things to consider:

1. We should also check for App mappings under `doctrine.orm.entity_manager.*`.
1. We could also add our App mapping under the default entity manager instead of globally.
1. We could check if `orm.auto_mapping` is enabled for the respective entity manager (or globally) and only add the mappings then.
1. Maybe we can reuse `DoctrineOrmMappingsPass`/`RegisterMappingsPass`.

----

Based on the things above I'd probably auto-add an App mapping to the default entity manager node if
1. the `<root>/src/Entity` folder exists
1. no mapping `App` is defined (globally) or for the default entity manager
1. `auto_mapping` is set to `true` for the default entity manager or globally

I'd also ignore all processed configuration.

@aschempp Wdyt?

Reply:
Sounds clear. However, we might should "hardcode" for the App name, but rather any name with the src/Entity path. Otherwise it would fail/double register if I use AppBundle for my name?

Reply:
> Otherwise it would fail/double register if I use AppBundle for my name?

Registering for the same dir twice doesn't seem to be a problem. This works in my test installation:
```yaml
doctrine:
  orm:
    mappings:
      App:
        type: annotation
        dir: '%kernel.project_dir%/src/Entity'
        is_bundle: false
        prefix: App\Entity
        alias: App
      AppBundle:
        type: annotation
        dir: '%kernel.project_dir%/src/Entity'
        is_bundle: false
        prefix: App\Entity
        alias: App
```